packages/block-editor/src/components/rich-text/native/get-format-colors.native.js

```diff
@@ -5,50 +5,43 @@ import { getColorObjectByAttributeValues } from '../../../components/colors';
 
 const FORMAT_TYPE = 'core/text-color';
 const REGEX_TO_MATCH = /^has-(.*)-color$/;
-const TAGS_TO_SEARCH = /\<mark/;
 
-export function getFormatColors( value, formats, colors ) {
-	if ( value?.search( TAGS_TO_SEARCH ) !== -1 ) {
-		const newFormats = formats.slice();
+export function getFormatColors( formats, colors ) {
+	const newFormats = formats.slice();
 
-		newFormats.forEach( ( format ) => {
-			format.forEach( ( currentFormat ) => {
-				if ( currentFormat?.type === FORMAT_TYPE ) {
-					const className = currentFormat?.attributes?.class;
-					currentFormat.attributes.style =
-						currentFormat.attributes.style.replace( / /g, '' );
```

Comment:
I'm a bit confused about the purpose of this line. On a first look, I thought we needed to reformat the style attribute by removing empty spaces. However, I see inconsistencies when applying a color and saving the post.

**After applying a text color:**
ℹ️ Note that the style doesn't have empty spaces.
```
<!-- wp:paragraph -->
<p><mark style="background-color:rgba(0, 0, 0, 0);color:#cf2e2e" class="has-inline-color has-vivid-red-color">Hello</mark></p>
<!-- /wp:paragraph -->
```

**After saving post:**
ℹ️ Note that the style has empty spaces.
```
<!-- wp:paragraph -->
<p><mark style="background-color: rgba(0, 0, 0, 0);color: #cf2e2e" class="has-inline-color has-vivid-red-color">Hello</mark></p>
<!-- /wp:paragraph -->
```

This behavior leads to an issue where the post is marked as modified every time you save it and select the text. I managed to reproduce this before these changes, so it's not a regression introduced in this PR. That said, seems this line can be removed.

cc @geriux as the [original author of this file](https://github.com/WordPress/gutenberg/pull/36028/files#diff-eeafa1b1a69c5e64e274f38913520c55fce6739f34789eb29724de657cdf33e0).

Reply:
I also though it was needed to do some checks below, but there's nothing actually checking the style as a whole, only the color value, so we should preserve the original styles and not modify anything.